.github/workflows/integration.yaml

```diff
@@ -70,3 +70,101 @@ jobs:
           go run ./cmd/wazero run -hostlogging=filesystem -mount=.:/ ${{ steps.zig-build-test-binary.outputs.ZIG_BINARY_PATH }}
         # TODO: remove when #1029 is complete
         continue-on-error: true
+
+  tinygo:
+    name: TinyGo
+    runs-on: ubuntu-20.04
+    env:
+      # Until https://github.com/tinygo-org/tinygo/pull/3399 is released in 0.27.0,
+      # use the artifact from the GitHub Actions on the TinyGo repository.
+      # 3961571741 matches the one of the merge commit for the PR ^.
+      # https://github.com/tinygo-org/tinygo/actions/runs/3961571741
+      TINYGO_RUN_ID: 3961571741
+
+    steps:
+      - uses: actions/cache@v3
+        id: cache
+        with:
+          path:
+            ~/.cache/go-build
+            ~/.cache/tinygo
+          key: integration-test-tinygo-${{ env.TINYGO_RUN_ID }}-${{ env.GO_VERSION }}
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Install TinyGo
+        if: steps.cache.outputs.cache-hit != 'true'
+        env:
+          # Necessary to use gh command.
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh run download --repo tinygo-org/tinygo ${{ env.TINYGO_RUN_ID }} -n linux-amd64-double-zipped
+          tar -xzf tinygo.linux-amd64.tar.gz
+          sudo dpkg -i tinygo_amd64.deb
+
+      - name: Checkout wazero
+        uses: actions/checkout@v3
+
+      - name: Install wazero
+        run: go install ./cmd/wazero
+
+      - name: Generate target json for wazero.
+        run: |
+          jq '.emulator = "wazero run -hostlogging=filesystem -mount=.:/ -mount={tmpDir}:/tmp {}"' \
+            $(tinygo env TINYGOROOT)/targets/wasi.json > ./wasi_wazero.json
+
+      - name: Run standard library tests
+        # On PRs, we only test the minimal std packages to test basic functionality.
+        run: tinygo test -target ./wasi_wazero.json os math hash
+
+      - name: Run all standard library tests
+        # Running all tests is pretty slow (~8min), so only do it on the main branch.
+        if: ${{ github.event_name  == 'push' }}
```

Comment:
already verified that this passes https://github.com/tetratelabs/wazero/actions/runs/3978659931/jobs/6820572028